pkg/controller/operators/olm/operatorconditions.go

```diff
@@ -0,0 +1,45 @@
+package olm
+
+import (
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorsv1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+func (a *Operator) isOperatorUpgradeable(csv *v1alpha1.ClusterServiceVersion) (bool, error) {
+	if csv == nil {
+		return false, fmt.Errorf("CSV is invalid")
+	}
+
+	cond, err := a.lister.OperatorsV1().OperatorConditionLister().OperatorConditions(csv.GetNamespace()).Get(csv.GetName())
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
+	}
+
+	// Check condition overrides
+	for _, override := range cond.Spec.Overrides {
```

Comment:
I think a better ux (requiring a webhook) would be to have `spec.overrides` actually change the value of `status` to match the overridden value, and then reject any external changes to status while the override is set (this is the part that requires a webhook).

Otherwise anyone else that is not OLM (including e.g. the UI) needs to re-implement this union logic. And the problem will grow as more supported conditions are added.

Reply:
Acked.